x-pack/plugins/siem/public/common/mock/global_state.ts

```diff
@@ -233,7 +232,7 @@ export const mockGlobalState: State = {
       },
     },
   },
-  alertList,
-  hostList,
-  management,
+  alertList: initialAlertListState as AlertListState,
```

Comment:
why do we have to cast here? I would have thought TS would derive the type

Reply:
TS knows the type, but the type is actually `Immutable<AlertListState>`. Our 'Immutable' types don't work with redux types exactly. 

Here's my approach for this PR: Inside each of the endpoint subplugins, state is Immutable. We use immutable reducer, immutable middleware, etc. But at the top level of the app, everything is cast back to the mutable version. This makes it so that the types will work correctly with things expecting a regular redux types.

I don't think the "top-level" parts of the app (the stuff that knows about subplugins, that sets up the store, etc) should know about our immutable stuff. 

For example, if you decided to do away w/ 'Immutable' stuff in 'management', it would be cool if you could do that without changing the 'top-level' code.

The immutable stuff isn't perfect, isn't strictly necessary, and should be something each subplugin opts into (or not.)

Let me know if that adds up.

Reply:
yeah, this makes sense. Thanks for the explanation